sphinx_immaterial/mathml_math_renderer.py

```diff
@@ -0,0 +1,40 @@
+"""Extension that enables docutils' MathML-based math rendering."""
+
+import docutils.nodes
+import docutils.utils.math
+from docutils.writers._html_base import HTMLTranslator as BaseHTMLTranslator
+from sphinx.application import Sphinx
+
+# Sphinx overrides Docutils' math rendering.  Here, we override Sphinx's
+# override to revert to docutils' math rendering.
+
+
+def visit_math(self: BaseHTMLTranslator, node: docutils.nodes.math):
+    self.math_output = "mathml"
+    self.math_output_options = []
```

Comment:
Should this be extended to user confval(s)?  Does this directly influence the docutils' converter used?

Reply:
This is basically equivalent to putting:

```
[html writers]
math_output: mathml
```

in a `docutils.conf` file next to the `conf.py` file.

There is one option that could be specified, an external LaTeX-syntax to MathML converter program to use in place of the one that is built into docutils.  Probably there should be a config option for that.

Alternatively this extension could be renamed "docutils_math_renderer" and we could require the user to create the separate docutils.conf file to specify the math format.  However, it seems unlikely that a user would want to use any of the docutils math formats other than mathml:
 - the `mathjax` format is already separately supported by sphinx, so I don't think there is any point in supporting that;
 - I don't know about the "HTML" format, but if we wanted to support that, we'd have to arrange to include the appropriate math.css stylesheet.

Reply:
I'm really just working with the [info from here](https://docutils.sourceforge.io/docs/user/config.html#math-output). I think it would be fine to only use the docutils converter script (latex2mathml). But I was thinking it might be useful if a user already has external converter installed (like pandoc or blahxml) and might want to use that instead.